src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -538,7 +555,40 @@ export class RoslynLanguageServer {
             childProcess = cp.spawn(serverPath, args, cpOptions);
         }
 
-        return childProcess;
+        // The server process will create the named pipe used for communcation. Wait for it to be created,
+        // and listen for the server to pass back the connection information via stdout.
+        const namedPipeConnectionPromise = new Promise<TransmittedInformation>((resolve) => {
+            childProcess.stdout.on('data', (data: { toString: (arg0: any) => any }) => {
+                const result: string = isString(data) ? data : data.toString(RoslynLanguageServer.encoding);
+                _channel.append('[stdout] ' + result);
+
+                // Use the regular expression to find all JSON lines
+                const jsonLines = result.match(RoslynLanguageServer.namedPipeKeyRegex);
+                if (jsonLines) {
+                    const transmittedPipeNameInfo: TransmittedInformation = JSON.parse(jsonLines[0]);
```

Comment:
I tried to do something reasonably simple here for a simple communication. Open to feedback or other ideas here.

Reply:
Resolved in Roslyn PR